pkg/gcp/actuator/actuator.go

```diff
@@ -493,52 +505,42 @@ func (a *Actuator) needsUpdate(ctx context.Context, cr *minterv1.CredentialsRequ
 	return serviceAPIsEnabled, false, nil
 }
 
-func (a *Actuator) buildRootGCPClient(cr *minterv1.CredentialsRequest) (ccgcp.Client, error) {
-	logger := a.getLogger(cr).WithField("secret", fmt.Sprintf("%s/%s", constants.CloudCredSecretNamespace, constants.GCPCloudCredSecretName))
+// buildReadGCPClient will return a GCP client using the scaled down read only GCP creds
+// for CCO, which are expected to live in openshift-cloud-credential-perator/cloud-credential-operator-gcp-ro-creds.
+// These creds would normally be created by CCO itself, via a CredentialsRequest created
+// while installing CCO.
+//
+// If these are not available but root creds are, we will use the root creds instead.
+// This allows us to create the read creds initially.
+func (a *Actuator) buildReadGCPClient(cr *minterv1.CredentialsRequest) (ccgcp.Client, error) {
+	logger := a.getLogger(cr).WithField("secret", fmt.Sprintf("%s/%s", roGCPCredsSecretNamespace, roGCPCredsSecret))
+	logger.Debug("loading GCP read-only credentials from secret")
 
-	logger.Debug("loading GCP credentials from secret")
-	jsonBytes, err := loadCredsFromSecret(a.Client, constants.CloudCredSecretNamespace, constants.GCPCloudCredSecretName)
+	jsonBytes, err := loadCredsFromSecret(a.Client, roGCPCredsSecretNamespace, roGCPCredsSecret)
 	if err != nil {
+		if errors.IsNotFound(err) {
+			logger.Warn("read-only creds not found, using root creds cllient")
+			return a.buildRootGCPClient(cr)
+		}
+		logger.WithError(err).Error("failed to load in read-only creds Secret")
 		return nil, err
 	}
 
-	logger.Debug("creating root GCP client")
+	logger.Debug("creating read GCP client")
 	return a.GCPClientBuilder(a.ProjectName, jsonBytes)
 }
```

Comment:
Looks like we also need to set upgradeable=false when that root cred is gone somewhere in the status area. Is it feasible to implement that now in the status handler in this branch, or would we be better waiting until the status controller merges?

Reply:
Good catch! I've put int he upgradeable() implementation to watch for the missing root creds, although one of these PRs might need some changes depending on which one gets in first.